src/test/java/com/atlassian/db/replica/api/TestDualConnection.java

```diff
@@ -1658,6 +1658,43 @@ public void shouldKeepUsingReplicaAfterSettingTimeout() throws SQLException {
             .containsOnly(REPLICA);
     }
 
+    @Test
+    public void shouldKeepUsingReplicaAfterSettingTimeoutUpperCase() throws SQLException {
+        final ConnectionProviderMock connectionProvider = new ConnectionProviderMock();
+        final Connection connection = DualConnection.builder(
+            connectionProvider,
+            permanentConsistency().build()
+        ).build();
+
+        try (Statement statement = connection.createStatement()) {
+            statement.execute("SET statement_timeout to 30000");
```

Comment:
If this is connected "statment timeouts leaks` then... 🤦 🤦

Reply:
It might be connected to the leaks if we used an upper case syntax. The other way it may be incorrect are different ways to call a query (executeUpdate, executeQuery etc.). The test case covers only `Statement#execute`.